plugins/utils/nvim-autopairs.nix

```diff
@@ -3,59 +3,130 @@
 
-    disabledFiletypes = mkOption {
-      type = types.nullOr (types.listOf types.str);
-      default = null;
-      description = "Disabled filetypes";
-    };
+      pairs = helpers.mkNullOrOption (types.attrsOf types.str) "Characters to pair up";
 
-    breakLineFiletypes = mkOption {
-      type = types.nullOr (types.listOf types.str);
-      default = null;
-      description = "Filetypes to break lines on";
-    };
+      disabledFiletypes =
+        helpers.defaultNullOpts.mkNullable
+        (types.attrsOf types.str)
+        "[ \"TelescopePrompt\" ]"
+        "Disabled filetypes";
 
-    htmlFiletypes = mkOption {
-      type = types.nullOr (types.listOf types.str);
-      default = null;
-      description = "Filetypes to treat as HTML";
-    };
+      # TODO remove this option at some point (it has been marked as removed since XX-XX-2023)
+      breakLineFiletypes = mkOption {
+        type = types.nullOr (types.listOf types.str);
+        default = null;
+        description = "Filetypes to break lines on";
+      };
+
+      # TODO remove this option at some point (it has been marked as removed since XX-XX-2023)
+      htmlFiletypes = mkOption {
+        type = types.nullOr (types.listOf types.str);
+        default = null;
+        description = "Filetypes to treat as HTML";
+      };
+
+      disableInMacro = helpers.defaultNullOpts.mkBool false ''
+        Disable when recording or executing a macro.
+      '';
+
+      disableInVisualblock = helpers.defaultNullOpts.mkBool false ''
+        Disable when insert after visual block mode.
+      '';
+
+      disableInReplaceMode = helpers.defaultNullOpts.mkBool true "Disable in replace mode.";
+
+      ignoredNextChar = helpers.defaultNullOpts.mkStr "[=[[%w%%%'%[%\"%.%`%$]]=]" ''
+        Regexp to ignore if it matches the next character.
+      '';
+
+      enableMoveright = helpers.defaultNullOpts.mkBool true "Enable moveright.";
+
+      enableAfterQuote = helpers.defaultNullOpts.mkBool true "Add bracket pairs after quote.";
 
-    ignoredNextChar = mkOption {
-      type = types.nullOr types.str;
-      default = null;
-      description = "Regexp to ignore if it matches the next character";
+      enableCheckBracketLine = helpers.defaultNullOpts.mkBool true "Check bracket in same line.";
+
+      enableBracketInQuote = helpers.defaultNullOpts.mkBool true "Enable bracket in quote.";
+
+      enableAbbr = helpers.defaultNullOpts.mkBool false "Trigger abbreviation.";
+
+      breakUndo = helpers.defaultNullOpts.mkBool true "Switch for basic rule break undo sequence.";
+
+      checkTs = helpers.defaultNullOpts.mkBool false "Use treesitter to check for a pair.";
+
+      tsConfig = helpers.mkNullOrOption (types.nullOr types.attrs) ''
+        Configuration for TreeSitter.
+
+        Default:
+        {
+          lua = [ "string" "source" ];
+          javascript = [ "string" "template_string" ];
+        };
```

Comment:
Wouldn't this need to be in \`\`\` to render correctly in the documentation

Reply:
Done :)